front/src/applications/operationalStudies/hooks/useEtcsBrakingCurves.ts

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import {
+  type EtcsBrakingCurve,
+  type EtcsBrakingCurves,
+  EtcsBrakingCurveType,
+  EtcsBrakingType,
+} from '@osrd-project/ui-charts';
+import { useSelector } from 'react-redux';
+
+import { useScenarioContext } from './useScenarioContext';
+import {
+  type EtcsBrakingCurvesResponse,
+  type EtcsCurves,
+  osrdEditoastApi,
+  type SimulationResponseSuccess,
+} from '../../../common/api/osrdEditoastApi';
+import { formatSpeedCurve } from '../../../modules/simulationResult/components/SpeedDistanceDiagram/helpers';
+import { findExceptionWithOccurrenceId } from '../../../modules/timetableItem/helpers/pacedTrain';
+import useSelectedTimetableItem from '../../../modules/timetableItem/hooks/useSelectedTimetableItem';
+import { getSelectedTrainId } from '../../../reducers/simulationResults/selectors';
+import { isPacedTrainResponseWithPacedTrainId, isTrainScheduleId } from '../../../utils/trainId';
+
+const formatEtcsCurves = (etcsBrakingCurves: EtcsBrakingCurvesResponse): EtcsBrakingCurves => {
+  const { conflicts, slowdowns, stops } = etcsBrakingCurves;
```

Comment:
nit (you can resolve if you're not convinced)
```suggestion
const formatEtcsCurves = ({ conflicts, slowdowns, stops }: EtcsBrakingCurvesResponse): EtcsBrakingCurves => {
```

Reply:
Sorry I prefer my version tbh, I'll just resolve this <3